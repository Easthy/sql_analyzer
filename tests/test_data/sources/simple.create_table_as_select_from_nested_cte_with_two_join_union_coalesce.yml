source.users:
  - id
  - name
  - project_id
source.user_status:
  - user_id
  - status
source.user_history:
  - user_id
  - name
  - status
  - project_id
source.imported_users:
  - user_id
  - name
  - status
  - project_id
reference.project:
  - project_id
  - name