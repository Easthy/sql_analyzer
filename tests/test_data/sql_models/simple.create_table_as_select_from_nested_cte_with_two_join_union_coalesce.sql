CREATE TABLE simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce AS
WITH _cte AS (
      WITH _nested_cte AS (
              SELECT users.id AS user_id,
                     users.name,
                     project.project_id
                FROM source.users

                     INNER JOIN reference.project
                     ON users.project_id = users.project_id
      )
      SELECT _nested_cte.*,
             user_status.status
        FROM _nested_cte

             LEFT JOIN source.user_status
             ON user_status.user_id = _nested_cte.user_id
)
SELECT _cte.user_id,
       _cte.name,
       _cte.status,
       _cte.project_id
  FROM _cte

UNION

SELECT COALESCE(user_history.user_id, imported_users.user_id) AS user_id,
       COALESCE(user_history.name, imported_users.name) AS name,
       COALESCE(user_history.status, imported_users.status) AS status,
       COALESCE(user_history.project_id, imported_users.project_id) AS project_id
  FROM source.user_history

       FULL OUTER JOIN source.imported_users
       ON imported_users.user_id = user_history.user_id
          AND imported_users.project_id = user_history.project_id
; 
