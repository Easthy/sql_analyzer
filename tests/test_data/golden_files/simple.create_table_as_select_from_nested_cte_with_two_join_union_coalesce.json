{
  "nodes": [
    { "id": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.name" },
    { "id": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.user_id" },
    { "id": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.status" },
    { "id": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.project_id" },
    { "id": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce" },
    { "id": "col:source.users.name" },
    { "id": "col:source.users.id" },
    { "id": "col:source.users.project_id" },
    { "id": "tbl:source.users" },
    { "id": "col:source.user_status.user_id" },
    { "id": "col:source.user_status.status" },
    { "id": "tbl:source.user_status" },
    { "id": "tbl:reference.project" },
    { "id": "col:reference.project.project_id" },
    { "id": "col:reference.project.name" },
    { "id": "tbl:source.user_history" },
    { "id": "col:source.user_history.user_id" },
    { "id": "col:source.user_history.name" },
    { "id": "col:source.user_history.status" },
    { "id": "col:source.user_history.project_id" },
    { "id": "tbl:source.imported_users" },
    { "id": "col:source.imported_users.user_id" },
    { "id": "col:source.imported_users.name" },
    { "id": "col:source.imported_users.status" },
    { "id": "col:source.imported_users.project_id" }
  ],
  "links": [
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.name", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.user_id", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.status", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.project_id", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "tbl:source.users", "type": "table_dependency" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "tbl:source.user_status", "type": "table_dependency" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "tbl:reference.project", "type": "table_dependency" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "tbl:source.imported_users", "type": "table_dependency" },
    { "source": "tbl:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce", "target": "tbl:source.user_history", "type": "table_dependency" },
    { "source": "tbl:source.users", "target": "col:source.users.name", "type": "contains_column" },
    { "source": "tbl:source.users", "target": "col:source.users.id", "type": "contains_column" },
    { "source": "tbl:source.users", "target": "col:source.users.project_id", "type": "contains_column" },
    { "source": "tbl:source.user_status", "target": "col:source.user_status.status", "type": "contains_column" },
    { "source": "tbl:source.user_status", "target": "col:source.user_status.user_id", "type": "contains_column" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.name", "target": "col:source.users.name", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.user_id", "target": "col:source.users.id", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.status", "target": "col:source.user_status.status", "type": "column_dependency" },
    { "source": "tbl:reference.project", "target": "col:reference.project.name", "type": "contains_column" },
    { "source": "tbl:reference.project", "target": "col:reference.project.project_id", "type": "contains_column" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.project_id", "target": "col:reference.project.project_id", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.project_id", "target": "col:source.user_history.project_id", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.user_id", "target": "col:source.user_history.user_id", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.name", "target": "col:source.user_history.name", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.status", "target": "col:source.user_history.status", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.project_id", "target": "col:source.imported_users.project_id", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.user_id", "target": "col:source.imported_users.user_id", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.name", "target": "col:source.imported_users.name", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_from_nested_cte_with_two_join_union_coalesce.status", "target": "col:source.imported_users.status", "type": "column_dependency" },
    { "source": "tbl:source.user_history", "target": "col:source.user_history.user_id", "type": "contains_column" },
    { "source": "tbl:source.user_history", "target": "col:source.user_history.name", "type": "contains_column" },
    { "source": "tbl:source.user_history", "target": "col:source.user_history.status", "type": "contains_column" },
    { "source": "tbl:source.user_history", "target": "col:source.user_history.project_id", "type": "contains_column" },
    { "source": "tbl:source.imported_users", "target": "col:source.imported_users.user_id", "type": "contains_column" },
    { "source": "tbl:source.imported_users", "target": "col:source.imported_users.name", "type": "contains_column" },
    { "source": "tbl:source.imported_users", "target": "col:source.imported_users.status", "type": "contains_column" },
    { "source": "tbl:source.imported_users", "target": "col:source.imported_users.project_id", "type": "contains_column" }
  ]
}